youtube_dl/YoutubeDL.py

```diff
@@ -861,6 +861,31 @@ def iter_playlistitems(format):
                     'extractor_key': ie_result['extractor_key'],
                 }
 
+                if self.params.get('playlistprompt', False):
+                    skipnextvid = False
+                    desc = entry.get('title', entry.get('id', ''))
+                    if not desc:
+                        entry_result = self.process_ie_result(entry,
```

Comment:
```process_ie_result``` is called twice. It's a big function and may contain heavy network I/O. For example:
```
$ youtube-dl -v "http://www.kuwo.cn/album/502294/" --geo-verification-proxy https://secure.uku.im:8443/ --playlist-prompt
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'http://www.kuwo.cn/album/502294/', '--geo-verification-proxy', 'https://secure.uku.im:8443/', '--playlist-prompt']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.03.22
[debug] Git HEAD: 093dad9e2
[debug] Python version 3.6.0 - Linux-4.10.4-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.2.4, ffprobe 3.2.4
[debug] Proxy map: {}
[kuwo:album] 502294: Download album info
[download] Downloading playlist: Made Series 《M》
[kuwo:album] playlist Made Series 《M》: Collected 1 video ids (downloading 1 of them)
[kuwo:song] 6430200: Download song detail info
[kuwo:song] 6430200: Download ape url info
[kuwo:song] 6430200: Download mp3-320 url info
[kuwo:song] 6430200: Download mp3-192 url info
[kuwo:song] 6430200: Download mp3-128 url info
[kuwo:song] 6430200: Download wma url info
[kuwo:song] 6430200: Download aac url info
[kuwo:song] 6430200: Download album detail info
Do you want to download video 1/1: "Loser"? (Y/n) y
[download] Downloading video 1 of 1
[kuwo:song] 6430200: Download song detail info
[kuwo:song] 6430200: Download ape url info
[kuwo:song] 6430200: Download mp3-320 url info
[kuwo:song] 6430200: Download mp3-192 url info
[kuwo:song] 6430200: Download mp3-128 url info
[kuwo:song] 6430200: Download wma url info
[kuwo:song] 6430200: Download aac url info
[kuwo:song] 6430200: Download album detail info
[debug] Invoking downloader on 'http://other.web.rf01.sycdn.kuwo.cn/60c92ace79968c55a3e53629a5bdec2d/58d2a746/resource/n1/71/58/2389040631.mp3'
[download] Destination: Loser-6430200.mp3
[download] 100% of 8.37MiB in 00:00
[download] Finished downloading playlist: Made Series 《M》
```
This is a geo-restricted album. In this example each "Download foobar url info" can take a few seconds if the proxy is slow. Sounds not good.

By the way, process_ie_result does not always return a dictionary. For example:
```
youtube-dl -vF "http://www.kuwo.cn/album/502294/" --geo-verification-proxy https://secure.uku.im:8443/ --playlist-prompt    
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-vF', 'http://www.kuwo.cn/album/502294/', '--geo-verification-proxy', 'https://secure.uku.im:8443/', '--playlist-prompt']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.03.22
[debug] Git HEAD: 093dad9e2
[debug] Python version 3.6.0 - Linux-4.10.4-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.2.4, ffprobe 3.2.4
[debug] Proxy map: {}
[kuwo:album] 502294: Download album info
[download] Downloading playlist: Made Series 《M》
[kuwo:album] playlist Made Series 《M》: Collected 1 video ids (downloading 1 of them)
[kuwo:song] 6430200: Download song detail info
[kuwo:song] 6430200: Download ape url info
[kuwo:song] 6430200: Download mp3-320 url info
[kuwo:song] 6430200: Download mp3-192 url info
[kuwo:song] 6430200: Download mp3-128 url info
[kuwo:song] 6430200: Download wma url info
[kuwo:song] 6430200: Download aac url info
[kuwo:song] 6430200: Download album detail info
[info] Available formats for 6430200:
format code  extension  resolution note
aac          aac        unknown    audio@ 48k
wma          wma        unknown    
mp3-128      mp3        unknown    audio@128k
mp3-192      mp3        unknown    audio@192k
mp3-320      mp3        unknown    audio@320k (best)
Traceback (most recent call last):
  File "<string>", line 23, in <module>
  File "/home/yen/Projects/youtube-dl/youtube_dl/__init__.py", line 465, in main
    _real_main(argv)
  File "/home/yen/Projects/youtube-dl/youtube_dl/__init__.py", line 455, in _real_main
    retcode = ydl.download(all_urls)
  File "/home/yen/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 1910, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/home/yen/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 773, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/yen/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 950, in process_ie_result
    desc = entry_result.get('title', entry_result.get('id', ''))
AttributeError: 'NoneType' object has no attribute 'get'
```